pkg/install/knative.go

```diff
@@ -32,45 +32,45 @@ import (
 
 const knativeAddressableResolverClusterRoleName = "addressable-resolver"
 
-// BindKnativeAddressableResolverClusterRole binds the Knative Addressable resolver aggregated ClusterRole
+// BindKnativeAddressableResolverClusterRole binds the Knative addressable resolver aggregated ClusterRole
 // to the operator ServiceAccount.
-func BindKnativeAddressableResolverClusterRole(ctx context.Context, c kubernetes.Interface, namespace string) error {
+func BindKnativeAddressableResolverClusterRole(ctx context.Context, c kubernetes.Interface, global bool, operatorNamespace string) error {
 	if isKnative, err := knative.IsInstalled(ctx, c); err != nil {
 		return err
 	} else if !isKnative {
 		return nil
 	}
-	if namespace != "" {
-		return applyAddressableResolverRoleBinding(ctx, c, namespace)
+	if global {
+		return applyAddressableResolverClusterRoleBinding(ctx, c, operatorNamespace)
 	}
-	return applyAddressableResolverClusterRoleBinding(ctx, c, namespace)
+	return applyAddressableResolverRoleBinding(ctx, c, operatorNamespace)
 }
 
-func applyAddressableResolverRoleBinding(ctx context.Context, c kubernetes.Interface, namespace string) error {
-	rb := rbacv1ac.RoleBinding(fmt.Sprintf("%s-addressable-resolver", serviceAccountName), namespace).
+func applyAddressableResolverRoleBinding(ctx context.Context, c kubernetes.Interface, operatorNamespace string) error {
+	rb := rbacv1ac.RoleBinding(fmt.Sprintf("%s-addressable-resolver", serviceAccountName), operatorNamespace).
```

Comment:
I realise I may have introduced the confusion in the first place, but I'm not sure this is going to work when the WATCH_NAMESPACE is different from the operator namespace. Is it? If not, we'll have to keep the WATCH_NAMESPACE argument.

Reply:
This may be a really rare edge case where a local operator is watching another foreign namespace. Not sure of this construct where WATCH_NAMESPACE and operator namespace are different even works. In fact the operator leader lease is then missing in this foreign namespace and other operators may also reconcile resources because of this.

Reply:
I think that's possible with OLM `SingleNamespace` installation mode that we support. The leader Lease if for making there is a single operator replica being active, it's not meant to be locking a namespace, even if it may have been diverted for that purpose :)

Reply:
@astefanutti I reverted to using both WATCH_NAMESPACE and operatorNamespace as an argument. Please have a final look. Many thanks